compiler/noirc_frontend/src/elaborator/expressions.rs

```diff
@@ -520,6 +586,40 @@ impl<'context> Elaborator<'context> {
         }
     }
 
+    /// Elaborates an expression taking into account that it's a call argument in a function
+    /// that has the given type, and `arg_index` is the index of that argument in that function type.
+    fn elaborate_call_argument_expression(
+        &mut self,
+        arg: Expression,
+        arg_index: usize,
+        func_type: &Type,
+        any_argument_has_lambda_without_type_annotations: bool,
```

Comment:
> any_argument_has_lambda_without_type_annotations

We can remove this flag from the PR (not just this method), it shouldn't be needed if we push down the type instead of unify afterward